compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/CoroutineCodegen.kt

```diff
@@ -144,6 +144,7 @@ private fun IrFunction.isStaticInlineClassReplacementDelegatingCall(): Boolean =
 
 private val BRIDGE_ORIGINS = setOf(
     IrDeclarationOrigin.FUNCTION_FOR_DEFAULT_PARAMETER,
+    JvmLoweredDeclarationOrigin.JVM_STATIC_WRAPPER,
```

Comment:
This is a little too subtle to me:

In JvmLower, line 271, we decide whether to insert special `Nothing` handling code. We do not do so for bridges. However, we make that determination using a suspend method helper method which uses this list of bridge origins. JVM_STATIC_WRAPPER was missing from the list which means that we insert the `Nothing` handling so that the wrapper becomes a block with two calls and not just the forwarding call it should be. That then trips up method flag calculation later.

The fact that this is using a suspend method utility also for non-suspend functions is a little puzzling. This is the minimal fix to the bug report, but we should probably separately try to make this more readable.

Reply:
The purpose of that condition in kotlinNothingValueExceptionPhase is precisely to handle suspend functions, as they technically *can* return a value (`COROUTINE_SUSPENDED`) so unless we decide to run the coroutine transformer later to handle that case, the exception-throwing code is not even valid.

On a related note, this code does not currently work:
```
import kotlin.coroutines.*
import kotlin.coroutines.intrinsics.*

@JvmInline
value class I(val x: Any?)

suspend fun <T> suspendHere(x: T): T = suspendCoroutineUninterceptedOrReturn {
    it.resume(x)
    COROUTINE_SUSPENDED
}

open class C {
    companion object {
        @JvmStatic
        protected suspend fun f(): I = I(suspendHere("OK"))
    }
}

class D : C() {
    companion object {
        suspend fun g() = f()
    }
}

suspend fun main() = println(D.g().x)
```
I thought that adding JVM_STATIC_WRAPPER to BRIDGE_ORIGINS would fix it, but unfortunately it only makes it fail in a different way a bit later.

Reply:
Thanks @pyos! I added your test case and the copying of attributes when creating the wrapper!